### PR TITLE
feat: [AH-3246] Calculate operational availability

### PR DESCRIPTION
**Pull request by VebjornG**
## Description
This is a function that gives a measure of how how long some equipment has been operational by using the mean up time and mean down time of the equipment in question. 
[Ticket](https://cognitedata.atlassian.net/browse/AH-3246?atlOrigin=eyJpIjoiNmJlNzMwNWZiNjI1NDQ2NTkyZTM0YzZiYTM5YzRlNDUiLCJwIjoiaiJ9)

## Screenshots

Screenshot 2024-06-13 at 15 27 33

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

## Contributor Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added an example of my new feature and included it in the documentation.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** [document](https://cognitedata.github.io/indsl/contribute.html).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My Pull Request name follows the [naming convention](https://www.conventionalcommits.org/en/v1.0.0/) `fix: <description>`, `feat: <description>`, etc.

## Reviewer Checklist for Charts compliant functions:

- [ ] The docstrings of the new function follow the contributing [guidelines](https://cognitedata.github.io/indsl/contribute.html).
- [ ] The new function is professionally [documented](https://journal.emwa.org/writing-matters/whats-your-problem-a-practical-approach-to-scientific-document-design/article/2110/2047480612z2e00000000042.pdf)
- [ ] The new function and associated scripts are covered by one or more unit tests and code coverage did not decrease.
- [ ] The new function is accompanied by an example and it is included in the Gallery of Charts.
- [ ] The new function is reviewed in Chromatic. Access the storybook build results url and comment, approve or deny.
- [ ] All function inputs, arguments, and outputs have a supported data type and have human readable names.
- [ ] No code language is included in the description of the function or parameters (e.g use "polynomial order" instead of "poly_order")

### CODE REVIEW

File: indsl/equipment/operational_availability.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    the mean up time (MUT) and mean down time (MDT). The formula for the operational
+    availability is given by
+
+    .. math::
+        A_o = \frac{MUT}{MUT + MDT}
+
+    where :math:`MUT` is the mean up time and :math:`MDT` is the mean down time.
+
+
+    Parameters
```
Comment (neringaalt) [inline indsl/equipment/operational_availability.py:17]: Please read indsl documentation on docstrings or look at any current function to check documentation style

File: indsl/equipment/operational_availability.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+
+
+def calculate_operational_availability(up_time_data: pd.Series, down_time_data: pd.Series) -> pd.Series:
+    r"""Calculate operational availability.
```
Comment (neringaalt) [inline indsl/equipment/operational_availability.py:5]: Thats the name that will be shown in Charts. I suppose we dont need `Calculate`?

File: indsl/equipment/operational_availability.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+
+
+def calculate_operational_availability(up_time_data: pd.Series, down_time_data: pd.Series) -> pd.Series:
```
Comment (neringaalt) [inline indsl/equipment/operational_availability.py:4]: Add `check_types` decorator

**Review by skepticalcat — DISMISSED** [review]
looks very good!

File: indsl/equipment/operational_availability_.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+
+from indsl.type_check import check_types
+
+
+@check_types
+def operational_availability(availability: pd.Series, output: str = "UT") -> pd.Series:
```
Comment (neringaalt) [inline indsl/equipment/operational_availability_.py:7]: output should be a `Literal`, not a string

Comment (neringaalt) [reply indsl/equipment/operational_availability_.py:7]: Can we also have full words?

File: indsl/equipment/operational_availability_.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import pandas as pd
```
Comment (neringaalt) [inline indsl/equipment/operational_availability_.py:1]: Always add `# Copyright 2024 Cognite AS` in every new file

**Review by skepticalcat — DISMISSED** [review]

File: indsl/equipment/operational_availability_.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    in a given time period.
+
+    Args:
+        availability: Availability.
+            Time series data of the availability of the system.
+        output: Output type.
+            A string representing the output of the function. Either 'Uptime' for uptime or 'Downtime' for downtime.
+
+    Returns:
+        pd.Series
```
Comment (neringaalt) [inline indsl/equipment/operational_availability_.py:26]: Add the name that we be shown in the UI

**Review by neringaalt — APPROVED** [review]

**Pull request merged**